R/process_sc_all_care_home.R

```diff
@@ -99,47 +89,58 @@ process_sc_all_care_home <- function(
 
 
   fixed_nursing_provision <- fixed_ch_provider %>%
-    dplyr::arrange(
-      "sending_location",
-      "social_care_id",
-      "period_start_date",
-      "ch_admission_date"
-    ) %>%
     dplyr::group_by(
       .data[["sending_location"]],
       .data[["social_care_id"]],
-      .data[["chi"]],
       .data[["ch_admission_date"]]
     ) %>%
     # fill in nursing care provision when missing
     # but present in the following entry
     dplyr::mutate(
       nursing_care_provision = dplyr::na_if(.data[["nursing_care_provision"]], 9L)
     ) %>%
-    tidyr::fill(all_of("nursing_care_provision"), .direction = "downup") %>%
-    dplyr::ungroup()
+    tidyr::fill(.data[["nursing_care_provision"]], .direction = "downup")
 
-  ready_to_merge <- fixed_nursing_provision %>%
-    #  dplyr::filter(chi == "3005291146") %>%
-    # remove any duplicate records before merging for speed and simplicity
-    dplyr::distinct() %>%
-    dplyr::arrange(
-      sending_location,
-      social_care_id,
-      period_start_date,
-      ch_admission_date
+
+  fixed_ch_provider <- fixed_nursing_provision %>%
+    dplyr::mutate(
+      ch_provider = dplyr::if_else(is.na(.data[["ch_provider"]]), 6L, .data[["ch_provider"]])
+    ) %>%
+    # sort data
+    # TODO - Different from SPSS. SPSS has nursing provider and period in the group_by. Needs investigation - does it matter?
+    dplyr::group_by(
+      .data[["sending_location"]],
+      .data[["social_care_id"]]
     ) %>%
     dplyr::group_by(
-      sending_location,
-      social_care_id,
-      chi,
-      ch_admission_date
+      .data[["sending_location"]],
+      .data[["social_care_id"]]
+    ) %>%
+    dplyr::mutate(
+      min_ch_provider = min(.data[["ch_provider"]]),
+      max_ch_provider = max(.data[["ch_provider"]]),
+      ch_provider = dplyr::if_else(
+        .data[["min_ch_provider"]] != .data[["max_ch_provider"]],
+        6L,
+        .data[["ch_provider"]]
+      )
     ) %>%
+    dplyr::select(
+      -"min_ch_provider",
+      -"max_ch_provider"
+    ) %>%
+    # tidy up ch_provider using 6 when disagreeing values
+    tidyr::fill(.data[["ch_provider"]], .direction = "downup") %>%
+    dplyr::ungroup()
```

Comment:
Comparing to master, these lines are duplicate code and the same thing is done on lines 60-88

Reply:
reverted back to master copy